ls1intum/Artemis
# Refactor Modeling Assessment Header into its own component

# Checklist

### Description
We want to unify the assessment ui for modeling and text exercises.
This PR extracts the header of the modeling assessment ui to its own component, so we can reuse it for text exercises.
Also, this adds extensive test coverage for the header.

➡️ We will reuse this component for Text Assessment (v2)

### Steps for Testing
1. Log in to Artemis
2. Navigate to a Modeling Exercise Assessment
3. See that the header works as before, no change should be visible.
    1. Assess a modeling submission yourself
    2. Open a modeling submission locked by someone else
    3. Override an existing modeling submission
